src/core/tests/type_prop/binary_elementwise.cpp

```diff
@@ -468,164 +468,175 @@ shared_ptr<op::v1::Reshape> createReshapeSubgraph(PartialShape param_shape,
-        const auto const_op = op::Constant::create(element::f32, {1}, {-10});
-        try {
-            const auto reshape = createReshapeSubgraph<op::v1::Add>(op_shape, const_op);
-            FAIL() << "Incorrect dimenion was not detected.";
-        } catch (const NodeValidationFailure& error) {
-            EXPECT_HAS_SUBSTRING(error.what(), std::string("Dim size cannot be less than -1"));
-        } catch (...) {
-            FAIL() << "Incorrect dimenion was not detected, different error type was thrown.";
-        }
+        FAIL() << "Incorrect dimension was not detected.";
+    } catch (const NodeValidationFailure& error) {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("Dim size cannot be less than -1"));
+    } catch (...) {
+        FAIL() << "Incorrect dimension was not detected, different error type was thrown.";
     }
 }
 
-TEST(type_prop, interval_value_propagation_sub) {
+TEST(type_prop, interval_value_propagation_sub_rhs) {
     PartialShape op_shape{Dimension(-1), Dimension(24, -1), Dimension(4, 36), Dimension(13, 27), Dimension(13, 27), 15};
-    {  // const rhs
-        const auto const_op = op::Constant::create(element::f32, {6}, {2, 3, 4, 5, -5, 6});
-        const auto reshape = createReshapeSubgraph<op::v1::Subtract>(op_shape, const_op);
-        EXPECT_EQ(reshape->get_element_type(), element::f32);
-        EXPECT_EQ(
-            reshape->get_output_partial_shape(0),
-            PartialShape({-1, /* Dimension(21, -1) */ -1, Dimension(-1, 32), Dimension(8, 22), Dimension(18, 32), 9}));
-    }
-    {  // const lhs
-        const auto const_op = op::Constant::create(element::f32, {6}, {12, 28, 36, 43, 27, 25});
+    // const rhs
+    const auto const_op = op::Constant::create(element::f32, {6}, {2, 3, 4, 5, -5, 6});
+    const auto reshape = createReshapeSubgraph<op::v1::Subtract>(op_shape, const_op);
+    EXPECT_EQ(reshape->get_element_type(), element::f32);
+    EXPECT_EQ(
+        reshape->get_output_partial_shape(0),
+        PartialShape({-1, /* Dimension(21, -1) */ -1, Dimension(-1, 32), Dimension(8, 22), Dimension(18, 32), 9}));
+}
+
+TEST(type_prop, interval_value_propagation_sub_lhs) {
+    PartialShape op_shape{Dimension(-1), Dimension(24, -1), Dimension(4, 36), Dimension(13, 27), Dimension(13, 27), 15};
+    // const lhs
+    const auto const_op = op::Constant::create(element::f32, {6}, {12, 28, 36, 43, 27, 25});
+    const auto reshape = createReshapeSubgraph<op::v1::Subtract>(op_shape, const_op, false);
+    EXPECT_EQ(reshape->get_element_type(), element::f32);
+    EXPECT_EQ(reshape->get_output_partial_shape(0),
+              PartialShape({-1, /* Dimension(8, -1) */ -1, Dimension(0, 32), Dimension(16, 30), Dimension(0, 14), 10}));
+}
+
+TEST(type_prop, interval_value_propagation_sub_incorrect_dim) {
+    // const lhs - result lower than 0
+    PartialShape op_shape{Dimension(13, 27)};
+    const auto const_op = op::Constant::create(element::f32, {1}, {5});
+    try {
```

Comment:
Totally optional: Consider make a copy of helper test checker from
https://github.com/openvinotoolkit/openvino/blob/12bbcb1b821f2c189c0b6959152512b31ff40d6f/src/tests/util/type_prop.hpp#L13

then assert check will simpler.

Reply:
Like it! I've cherry picked the commit with the helper as the PR is not merged yet.
Thank you:)